File: Apps/BankBusiness/backend/model/ContactPerson.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.model;

[Table("contact_person")]
public class ContactPerson
{
    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id {get; set;}
    
    [Column(name: "surname")]
    public string Surname {get; set;}
    
    [Column(name: "name")]
    public string Name {get; set;}
    
    [Column(name: "patronymic")]
    public string Patronymic {get; set;}
    
    [Column(name: "phone_number")]
    public string PhoneNumber {get; set;}
    
    [Column(name: "email")]
    public string Email {get; set;}
    
    [Column(name: "business_client_id"), NotMapped]
    public BusinessClient LinledClient {get; set;}
}

File: Apps/BankBusiness/backend/src/main/scala/model/ContactPerson.scala
package model

import io.getquill.{InsertMeta, SchemaMeta, UpdateMeta, insertMeta, schemaMeta, updateMeta}
import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}

case class ContactPerson(
                          id: Long,
                          surname: String,
                          name: String,
                          patronymic: String,
                          phoneNumber: String,
                          email: String,
                          businessClientId: Long
                        )

object ContactPerson {
  inline given SchemaMeta[ContactPerson] = schemaMeta(
    "contact_person",
    _.id -> "id",
    _.surname -> "surname",
    _.name -> "name",
    _.patronymic -> "patronymic",
    _.phoneNumber -> "phone_number",
    _.email -> "email",
    _.businessClientId -> "business_client_id",
  )

  inline given InsertMeta[ContactPerson] = insertMeta(_.id)

  inline given UpdateMeta[ContactPerson] = updateMeta(_.id)

  implicit val encoder: JsonEncoder[ContactPerson] = DeriveJsonEncoder.gen[ContactPerson]
  implicit val decoder: JsonDecoder[ContactPerson] = DeriveJsonDecoder.gen[ContactPerson]
}